extensions/samples/electron-webpack/src/AuthProvider.ts

```diff
@@ -0,0 +1,162 @@
+import path from "path";
+
+export default class AuthProvider {
+    private clientApplication: PublicClientApplication;
+    private account: AccountInfo;
+
+    constructor() {}
+
+    static async build(authConfig: any): Promise<any> {
+        try {
+            const authProvider = new AuthProvider();
+            authProvider.account = null;
+            const cachePath = path.join(
+                Environment.getUserRootDirectory(),
+                "./cache.json"
+            );
+
+            const persistenceConfiguration = {
+                cachePath,
+                dataProtectionScope: DataProtectionScope.CurrentUser,
+                serviceName: "test-msal-electron-service",
+                accountName: "test-msal-electron-account",
+                usePlaintextFileOnLinux: false,
+            };
+
+            const persistence = await PersistenceCreator.createPersistence(
+                persistenceConfiguration
+            );
+            authProvider.clientApplication = new PublicClientApplication({
+                auth: authConfig.auth,
+                // This hooks up the cross-platform cache into MSAL
+                cache: {
+                    cachePlugin: new PersistenceCachePlugin(persistence),
+                },
+                system: {
+                    loggerOptions: {
+                        loggerCallback(loglevel, message, containsPii) {
+                            console.log(message);
+                        },
+                        piiLoggingEnabled: false,
+                        logLevel: LogLevel.Info,
+                    },
+                },
+            });
+            return authProvider;
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async login(): Promise<AccountInfo> {
+        const tokenRequest: SilentFlowRequest = {
+            scopes: [],
+            account: null,
+        };
+        const authResponse = await this.getToken(tokenRequest);
+        return this.handleResponse(authResponse);
+    }
+
+    async getToken(
+        tokenRequest: SilentFlowRequest
+    ): Promise<AuthenticationResult> {
+        try {
+            let authResponse: AuthenticationResult;
+            const account = this.account || (await this.getAccount());
+            if (account) {
+                tokenRequest.account = account;
+                return (authResponse = await this.getTokenSilent(tokenRequest));
+            } else {
+                return (authResponse = await this.getTokenInteractive(
+                    tokenRequest
+                ));
+            }
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async getTokenSilent(
+        tokenRequest: SilentFlowRequest
+    ): Promise<AuthenticationResult> {
+        try {
+            return await this.clientApplication.acquireTokenSilent(
+                tokenRequest
+            );
+        } catch (error) {
+            console.log(
```

Comment:
should check first if error is `InteractionRequiredAuthError`

Reply:
Done